### PR TITLE
[Net] Rename "ssl" references to "tls" in methods and members.

### PR DESCRIPTION
**Pull request by Faless**
Follow-up on #65447 .
Also convert most references from SSL to TLS in docs.

CC @godotengine/documentation :

The tutorial URL also needs renaming in godot-docs.

We probably want to review if/where to use `SSL/TLS` over just `TLS` in the docs.

I feel like it make them clearer, but we might want to only use the combo in class descriptions? Or maybe just the HTTP* classes and methods (the web is the only place where "SSL" is still often wrongly used as a synonym to "TLS").

### CODE REVIEW

**Review by fire — APPROVED** [review]
I support the replacement of terminology of SSL with TLS.

File: modules/websocket/doc_classes/WebSocketServer.xml [hunk 1]
```diff
@@ -79,15 +79,15 @@
 			When not set to [code]*[/code] will restrict incoming connections to the specified IP address. Setting [code]bind_ip[/code] to [code]127.0.0.1[/code] will cause the server to listen only to the local host.
 		</member>
 		<member name="ca_chain" type="X509Certificate" setter="set_ca_chain" getter="get_ca_chain">
-			When using SSL (see [member private_key] and [member ssl_certificate]), you can set this to a valid [X509Certificate] to be provided as additional CA chain information during the SSL handshake.
+			When using SSL (see [member private_key] and [member tls_certificate]), you can set this to a valid [X509Certificate] to be provided as additional CA chain information during the SSL handshake.
```
Comment (Calinou) [inline modules/websocket/doc_classes/WebSocketServer.xml:82]: This still mentions SSL by name.

File: modules/websocket/doc_classes/WebSocketServer.xml [hunk 2]
```diff
@@ -79,15 +79,15 @@
...
 		</member>
 		<member name="handshake_timeout" type="float" setter="set_handshake_timeout" getter="get_handshake_timeout" default="3.0">
 			The time in seconds before a pending client (i.e. a client that has not yet finished the HTTP handshake) is considered stale and forcefully disconnected.
 		</member>
 		<member name="private_key" type="CryptoKey" setter="set_private_key" getter="get_private_key">
-			When set to a valid [CryptoKey] (along with [member ssl_certificate]) will cause the server to require SSL instead of regular TCP (i.e. the [code]wss://[/code] protocol).
+			When set to a valid [CryptoKey] (along with [member tls_certificate]) will cause the server to require SSL instead of regular TCP (i.e. the [code]wss://[/code] protocol).
```
Comment (Calinou) [inline modules/websocket/doc_classes/WebSocketServer.xml:88]: This still mentions SSL by name.

File: modules/websocket/doc_classes/WebSocketClient.xml [hunk 3]
```diff
@@ -50,11 +50,11 @@
 		</method>
 	</methods>
 	<members>
-		<member name="trusted_ssl_certificate" type="X509Certificate" setter="set_trusted_ssl_certificate" getter="get_trusted_ssl_certificate">
+		<member name="trusted_tls_certificate" type="X509Certificate" setter="set_trusted_tls_certificate" getter="get_trusted_tls_certificate">
 			If specified, this [X509Certificate] will be the only one accepted when connecting to an SSL host. Any other certificate provided by the server will be regarded as invalid.
-			[b]Note:[/b] Specifying a custom [code]trusted_ssl_certificate[/code] is not supported in Web exports due to browsers restrictions.
+			[b]Note:[/b] Specifying a custom [code]trusted_tls_certificate[/code] is not supported in Web exports due to browsers restrictions.
```
Comment (Calinou) [inline modules/websocket/doc_classes/WebSocketClient.xml:55]: Small typo fix:

```suggestion
			[b]Note:[/b] Specifying a custom [code]trusted_tls_certificate[/code] is not supported in Web exports due to browsers' restrictions.
```

File: doc/classes/HTTPClient.xml [hunk 4]
```diff
@@ -7,17 +7,17 @@
...
 		See the [HTTPRequest] node for a higher-level alternative.
 		[b]Note:[/b] This client only needs to connect to a host once (see [method connect_to_host]) to send multiple requests. Because of this, methods that take URLs usually take just the part after the host instead of the full URL, as the client is already connected to a host. See [method request] for a full example and to get started.
-		A [HTTPClient] should be reused between multiple requests or to connect to different hosts instead of creating one client per request. Supports SSL and SSL server certificate verification. HTTP status codes in the 2xx range indicate success, 3xx redirection (i.e. "try again, but over here"), 4xx something was wrong with the request, and 5xx something went wrong on the server's side.
+		A [HTTPClient] should be reused between multiple requests or to connect to different hosts instead of creating one client per request. Supports TLS and TLS server certificate verification. HTTP status codes in the 2xx range indicate success, 3xx redirection (i.e. "try again, but over here"), 4xx something was wrong with the request, and 5xx something went wrong on the server's side.
```
Comment (Calinou) [inline doc/classes/HTTPClient.xml:10]: I suggest rewording it to sound more natural, and also spell out the first occurrence of the acronym:

```suggestion
		A [HTTPClient] should be reused between multiple requests or to connect to different hosts instead of creating one client per request. Supports Transport Layer Security (TLS), including server certificate verification. HTTP status codes in the 2xx range indicate success, 3xx redirection (i.e. "try again, but over here"), 4xx something was wrong with the request, and 5xx something went wrong on the server's side.
```

**Review by akien-mga — APPROVED** [review]
Looks good to me.
